txp/Pos/Txp/Logic/Global.hs

```diff
@@ -134,5 +133,4 @@ runToilAction action = runDBToil . runToilTGlobal $ action
 
 -- Zip block's TxAuxes and corresponding TxUndos.
 blundToAuxNUndo :: TxpBlund -> [(TxAux, TxUndo)]
-blundToAuxNUndo (Left _, _)                = []
-blundToAuxNUndo (Right (_, payload), undo) = zip (flattenTxPayload payload) undo
+blundToAuxNUndo (payload, undo) = zip (flattenTxPayload (bcmPayload payload)) (bcmPayload undo)
```

Comment:
I think it's bad, because if payload is `ComponentBlockGenesis`, it will be runtime error, AFAIU. You need to pattern-match.